pkg/hooks/hooks.go

```diff
@@ -82,14 +99,18 @@ func (h *hookContext) Close() error {
 //
 // NB: in the event of a timeout this will leak descriptors
 func (h *HookExecContext) RunWithTimeout(logger logging.Logger) error {
-	finished := make(chan struct{})
-	go func() {
-		h.Run(logger)
+	finished := make(chan RunHookResult)
+	go func(r chan RunHookResult) {
+		err := h.Run(logger)
+		r <- RunHookResult{err}
 		close(finished)
```

Comment:
seems better to add `defer close(finished)` after the line `finished := make(chan RunHookResult)`

Reply:
I added a `defer` however I needed to add it in the channel func. When the defer was outside I would see panics when a hook would timeout stating that the channel was closed.